provider/resource_keycloak_group_roles.go

```diff
@@ -111,21 +116,33 @@ func resourceKeycloakGroupRolesReconcile(data *schema.ResourceData, meta interfa
 		return err
 	}
 
-	// remove roles
-	err = removeRolesFromGroup(keycloakClient, updates.clientRolesToRemove, updates.realmRolesToRemove, group)
-	if err != nil {
-		return err
+	if data.Get("exhaustive").(bool) {
```

Comment:
I think it could be a bit confusing when you delete a role_id from your `keycloak_group_roles`-resource with `exhaustive=false`. On a `terraform plan` terraform will show that the role will be deleted, but when you apply it, the role is still there. 

I solved it in the `keycloak_user_groups` resource by deleting all resource changes in any way. https://github.com/mrparkers/terraform-provider-keycloak/pull/505/files#diff-128eb8509bf9825644f6e772b728791c91d30754c5ddeea91407c52a09c09290R81